systemtest/src/test/java/io/strimzi/systemtest/upgrade/MappedYamlArgumentsProvider.java

```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.upgrade;
+
+public class MappedYamlArgumentsProvider {
+}
```

Comment:
I am just curious. What is the semantics of these files inside the system-test module?

Reply:
Ah, damn! I committed this by accident.

For context, I was trying to add assertions to the update ST and I found I didn't really like the `@JsonFileResource` as a way to supply arguments to the parameterised test, because:

1. There's no documentation of the fields, so it's difficult to come along and figure out how to write the JSON to test a particular scenario.
2. If you're only interested in your scenario you have to run the test with all the combinations before you can pick out the individual scenario you added.

So I was messing about with writing something which functioned a lot like `@JsonFileResource` but:

* Used YAML rather than JSON (thus supporting comments, solving problem 2).
* Uses the Jackson YAMLMapper to provide typed arguments to the test method, with classes for the `Upgrade`, `Image`s, `EnvironmentInfo` etc. This would proved a place to document what all those properties are for. 

Another advantage would be extra type safety in those tests.

Anyway, I didn't open a PR for this because I felt it would be a time sink to push it through, but I still believe it would be an improvement.